File: PAT_Basic_Level_by_cpp/1006 换个格式输出整数/1006 换个格式输出整数.cpp
// 1006 换个格式输出整数
// 用 B 表示"百"、S 表示"十"，用 123...n 表示个位数字 n（n < 10）。
// 输入为不超过 3 位的正整数。
//
// 【实现原理】
// 算法思路：
//   将三位数的百位、十位、个位分别提取出来，按题目要求格式化输出：
//   - 百位数字决定了输出多少个 'B'
//   - 十位数字决定了输出多少个 'S'
//   - 个位数字决定了输出 1 到该数字的连续整数序列
// 关键点：
//   - 不足三位时，高位默认为 0，对应的循环不执行，不会输出任何字符。
// 时间复杂度：O(n)，其中 n 为输入数字的位数（最多 3 位），可视为 O(1)。
// 空间复杂度：O(1)。
#include <iostream>

using namespace std;

int main() {
    int n;
    if (!(cin >> n)) return 0;

    int hundreds = n / 100;
    int tens = (n / 10) % 10;
    int ones = n % 10;

    for (int i = 0; i < hundreds; ++i) cout << 'B';
    for (int i = 0; i < tens; ++i) cout << 'S';
    for (int i = 1; i <= ones; ++i) cout << i;

    cout << endl;
    return 0;
}

File: PAT_Basic_Level_by_lua/1006 换个格式输出整数/1006 换个格式输出整数.lua
-- 题目编号: 1006 换个格式输出整数
-- 实现原理: 将三位整数的百位、十位、个位分别提取，百位用若干个'B'表示，十位用若干个'S'表示，个位用从1到个位数的数字序列表示，拼接后输出。

-- 读取输入的正整数n（不超过1000）
local n = tonumber(io.read())
-- 用于存储输出结果的数组
local result = {}

-- 提取百位数：除以100再取整后对10取余
local hundreds = math.floor(n / 100) % 10
-- 提取十位数：除以10再取整后对10取余
local tens = math.floor(n / 10) % 10
-- 提取个位数：直接对10取余
local ones = n % 10

-- 在结果中添加百位对应的B字符（hundreds个B）
result[#result + 1] = string.rep("B", hundreds)
-- 在结果中添加十位对应的S字符（tens个S）
result[#result + 1] = string.rep("S", tens)
-- 在结果中添加个位对应的数字序列（从1到ones）
for i = 1, ones do
    result[#result + 1] = tostring(i)
end

-- 将结果数组拼接成字符串并输出
print(table.concat(result))
